src/java/rems/InvalidRequestException.java

```diff
@@ -0,0 +1,26 @@
+package rems;
+
+/**
+ * Thrown when a request is invalid.
+ */
+public class InvalidRequestException extends Exception {
+
+    public InvalidRequestException() {
+    }
+
+    public InvalidRequestException(String message) {
+        super(message);
+    }
+
+    public InvalidRequestException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public InvalidRequestException(Throwable cause) {
+        super(cause);
+    }
+
+    public InvalidRequestException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace) {
```

Comment:
This seems to have `protected` modifier in Exception class, should it be protected here as well? See https://docs.oracle.com/javase/7/docs/api/java/lang/Exception.html

Reply:
Ok, in that case the constructor can be removed.

Reply:
Hm, yeah, true.